decidim/decidim
# Clean meetings form with registrations

## :tophat: What? Why?

There were a couple of inconsistencies between the meeting admin form and the meeting registrations admin form, such as:

* when registrations are "on this platform", we have the available_slots in both forms
* with any kind of registrations (even registrations disabled), you can change the  custom email registration on the meeting form. This field should only be changed  when registrations are "on this platform", 
* when registrations are "external platform", there's a disclaimer without styling
* while working on these things, I found in a JavaScript of the meeting form a reference to meeting_registration_terms that was migrated to the registrations form sometime ago

This PR fixes all of these issues. I could have made it granular, but I think that would complicate the merge as we would have conflicts everywhere.

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #7416
- Fixes  #8267

## Testing

1. Create or edit meeting in the admin form with registrations "on this platform", see that you don't have the available slots nor the email customization fields. See that those fields are in the registrations form. 
2. Create or edit meeting in the admin form with registrations "on an external platform", see that you have styling in the disclaimer

## :camera: Screenshots

|Before | After|
|--------|------|
| ![image](https://user-images.githubusercontent.com/717367/140531906-a29be492-a0c3-427f-a28e-492b73a5f3c4.png) | ![image](https://user-images.githubusercontent.com/717367/140530763-641edc58-b02f-4b7d-ab31-2ecde91932e1.png) | 
| ![image](https://user-images.githubusercontent.com/717367/140531918-166eecea-f895-4f51-81be-7e93e6ec7e8f.png) | ![image](https://user-images.githubusercontent.com/717367/140530770-b0178a78-4697-4025-9e0d-af5908ed06dd.png) |
|  ![image](https://user-images.githubusercontent.com/717367/140531925-4252c000-8b47-45a3-9016-7dff616704d7.png) |  ![image](https://user-images.githubusercontent.com/717367/140530777-c4629ff3-fd06-4dcd-981a-f39d04bf6d99.png) ![image](https://user-images.githubusercontent.com/717367/140530790-d9b9ef45-18f3-4793-8b69-02cabfeaf72c.png) | 


:hearts: Thank you!
